### PR TITLE
Extracted ViewInterface from yii\web\View yiisoft/yii2#14997

### PR DESCRIPTION
**Pull request by njasm**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14997

### CODE REVIEW

File: framework/web/ViewInterface.php [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+     * @return string the rendering result
+     * @see render()
+     */
+    public function renderAjax($view, $params = [], $context = null);
+
+    /**
+     * Registers the asset manager being used by this view object.
+     * @return \yii\web\AssetManager the asset manager. Defaults to the "assetManager" application component.
+     */
+    public function getAssetManager();
```
Comment (samdark) [inline framework/web/ViewInterface.php]: Doesn't look like it belongs to the same interface...

File: framework/web/ViewInterface.php [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+    /**
+     * Sets the asset manager.
+     * @param \yii\web\AssetManager $value the asset manager
+     */
+    public function setAssetManager($value);
```
Comment (samdark) [inline framework/web/ViewInterface.php]: Doesn't look like it belongs to the same interface...

**Comment by njasm** [comment]
this pull is not finished yet.

We do have a base view that I still didn't understand why, the only usage (or reference to it) is the `yi\web\View`that extend it.

I believe the main idea was to have a base View where the Web and Console controller could have they're own view implementations [plus the base methods that are used by the framework that probably are not in the context of Web Controller or Console], don't know if this is true. need to investigate further.

**Comment by samdark** [comment]
Yes, your assumption is true.

**Pull request closed without merging**